security/filter/security-filter-login/src/main/java/org/codice/ddf/security/filter/login/LoginFilter.java

```diff
@@ -105,12 +105,24 @@
 
     private Crypto signatureCrypto;
 
-    private DocumentBuilder docBuilder;
-
     private Validator assertionValidator = new SamlAssertionValidator();
 
     private final Object lock = new Object();
 
+    private static final ThreadLocal<DocumentBuilder> BUILDER =
+            new ThreadLocal<DocumentBuilder>() {
+                @Override
+                protected DocumentBuilder initialValue() {
+                    try {
+                        DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
+                        factory.setNamespaceAware(true);
+                        return factory.newDocumentBuilder();
+                    } catch (ParserConfigurationException ex) {
+                        throw new IllegalArgumentException("Unable to create new DocumentBuilder", ex);
+                    }
```

Comment:
Shouldn't we log something here? Also, since this would leave the BUILDER null, do we need to check for null when calling BUILDER.get(), or return some kind of null object that logs and throws a SecurityServiceException or InvalidStateException?